test/tests/functional/pbs_job_default_group.py

```diff
@@ -56,12 +54,22 @@ def test_job_with_default_group(self):
         """
         if self.server.hostname == self.mom.hostname:
             self.skipTest("Server and Execution host must be different")
-        # add a temporary new user on execution host
-        self.user_name = "temppbs" + \
-            "".join(random.choices(string.ascii_letters, k=5))
+        # add a temporary new user on execution host and assume that user
+        # will be created and check user is not present on server host
+        self.user_name = "ptlpbstestuser1"
+        try:
+            pwd.getpwnam(self.user_name)
+            self.skipTest(f"{self.user_name} should not be present "
```

Comment:
I'm not sure we want to skip the test just because the user_name is present.  Once we get into the situation, then the test would always be skipped and this feature wouldn't be tested.
Since you know the test created the user, can the test just remove the user and then add it again?
If it isn't possible to delete the user, then the test should fail, not skip.

Reply:
@vishwaks or as I suggested earlier, if user exist, then modify it to take out the default group. Then you can proceed with the test.

Reply:
@nishiya and @bayucan :
Test is creating the user on execution host (which should be not server host), here I'm making sure that user is not present on the server host (place where PTL runs).

@bayucan : the '-default-' group which PBS assign is a virtual group assigned by PBS.

Reply:
@vishwaks I see.  The test won't run properly if the user exists on the server node.  But we would still have the problem of the test skipping and the change not getting tested.  Could we delete the user on the server node instead of skipping the test? What do you think?

Reply:
When we skip it, the test runner can see it and take action (delete the user on server host). Again run the test.
I'm okay to delete the user.

Reply:
Thanks @vishwaks.  Although in theory what you say is true, and could be done...in practice, I doubt most folks look at why a test was skipped...they only care that it is not a failure.  I think deleting the user will be the best course of action.